File: mvp/sdk/python/saas_ia/api/image_gen.py
"""Image Gen API — AI image generation (10 styles) + Real-ESRGAN upscaling."""

from __future__ import annotations

from typing import Any

from saas_ia.api.base import BaseAPI


class ImageGenAPI(BaseAPI):
    """Wraps ``/api/images`` endpoints."""

    async def generate(
        self,
        prompt: str,
        *,
        style: str | None = None,
        width: int | None = None,
        height: int | None = None,
    ) -> dict[str, Any]:
        """Generate an image from a prompt."""
        body: dict[str, Any] = {"prompt": prompt}
        if style:
            body["style"] = style
        if width:
            body["width"] = width
        if height:
            body["height"] = height
        return await self._post("/api/images/generate", json=body)

    async def thumbnail(
        self,
        title: str,
        *,
        style: str | None = None,
        transcription_id: str | None = None,
    ) -> dict[str, Any]:
        """Generate a YouTube thumbnail."""
        body: dict[str, Any] = {"title": title}
        if style:
            body["style"] = style
        if transcription_id:
            body["transcription_id"] = transcription_id
        return await self._post("/api/images/thumbnail", json=body)

    async def bulk(
        self,
        prompts: list[dict[str, Any]],
    ) -> list[dict[str, Any]]:
        """Bulk generate multiple images."""
        return await self._post("/api/images/bulk", json={"prompts": prompts})

    async def list(
        self,
        *,
        limit: int | None = None,
        offset: int | None = None,
    ) -> list[dict[str, Any]]:
        """List generated images."""
        return await self._get(
            "/api/images/",
            params={"limit": limit, "offset": offset},
        )

    async def delete(self, image_id: str) -> None:
        """Delete an image."""
        await self._delete(f"/api/images/{image_id}")

    async def upscale(self, image_id: str) -> dict[str, Any]:
        """Upscale an image with Real-ESRGAN."""
        return await self._post(f"/api/images/{image_id}/upscale")

    async def list_styles(self) -> list[dict[str, Any]]:
        """List available image styles."""
        return await self._get("/api/images/styles")

File: mvp/sdk/typescript/src/api/image-gen.ts
import { BaseAPI } from "./base";
import {
  GeneratedImage,
  ImageGenerateRequest,
  ImageThumbnailRequest,
  ImageBulkRequest,
  ImageStyle,
  PaginationParams,
} from "../types";

/**
 * Image Gen API — AI image generation (10 styles) with optional
 * Real-ESRGAN upscaling.
 */
export class ImageGenAPI extends BaseAPI {
  // -- Generation ----------------------------------------------------------

  /** Generate an image from a prompt. */
  async generate(data: ImageGenerateRequest): Promise<GeneratedImage> {
    return this._post<GeneratedImage>("/api/images/generate", data);
  }

  /** Generate a YouTube thumbnail. */
  async thumbnail(data: ImageThumbnailRequest): Promise<GeneratedImage> {
    return this._post<GeneratedImage>("/api/images/thumbnail", data);
  }

  /** Bulk generate multiple images. */
  async bulk(data: ImageBulkRequest): Promise<GeneratedImage[]> {
    return this._post<GeneratedImage[]>("/api/images/bulk", data);
  }

  // -- CRUD ----------------------------------------------------------------

  /** List generated images. */
  async list(params?: PaginationParams): Promise<GeneratedImage[]> {
    return this._get<GeneratedImage[]>("/api/images/", {
      limit: params?.limit,
      offset: params?.offset,
    });
  }

  /** Delete an image. */
  async delete(id: string): Promise<void> {
    return this._delete(`/api/images/${id}`);
  }

  // -- Upscale -------------------------------------------------------------

  /** Upscale an image with Real-ESRGAN. */
  async upscale(id: string): Promise<GeneratedImage> {
    return this._post<GeneratedImage>(`/api/images/${id}/upscale`);
  }

  // -- Projects ------------------------------------------------------------

  /** Create an image project. */
  async createProject(data: { name: string; description?: string }): Promise<Record<string, unknown>> {
    return this._post("/api/images/projects", data);
  }

  /** List image projects. */
  async listProjects(params?: PaginationParams): Promise<Record<string, unknown>[]> {
    return this._get("/api/images/projects", {
      limit: params?.limit,
      offset: params?.offset,
    });
  }

  // -- Reference -----------------------------------------------------------

  /** List available image styles. */
  async listStyles(): Promise<Array<{ id: ImageStyle; name: string; description: string }>> {
    return this._get("/api/images/styles");
  }
}
